### PR TITLE
Return KroneckerDelta when alpha,beta,gamma are 0 in WignerD Function

### LINKED ISSUE #6521: WignerD with angles all 0 should return delta function
When all the angles passed to the Wigner-D function are zero, it would be nice if it returned a KroneckerDelta, rather than a Wigner small-d function. Currently it gives:

```
In [1]: from sympy.physics.quantum.spin import WignerD

In [2]: j, m, mp = symbols('j m mp')

In [3]: WignerD(j, m, mp, 0, 0, 0)
Out[3]: 
  j     
d    (0)
 m,mp   
```

It's easy to do a `if args[3:] == (0, 0, 0):` in the `__new__` function and return a KroneckerDelta:

```
In [4]: WignerD(j, m, mp, 0, 0, 0)
Out[4]: 
δ
 m,mp
```

But just doing this will mess up some other stuff with the rewrite machinery:

```
In [5]: from sympy.physics.quantum.spin import JzKet

In [6]: JzKet(j, m).rewrite('Jz')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-19-6ad2c34dbd31> in <module>()
----> 1 JzKet(j, m).rewrite('Jz')

/home/sean/sympy/sympy/core/basic.pyc in rewrite(self, *args, **hints)
   1386 
   1387             if not pattern:
-> 1388                 return self._eval_rewrite(None, rule, **hints)
   1389             else:
   1390                 if iterable(pattern[0]):

/home/sean/sympy/sympy/physics/quantum/qexpr.pyc in _eval_rewrite(self, pattern, rule, **hints)
    330         if pattern is None or isinstance(self, pattern):
    331             if hasattr(self, rule):
--> 332                 rewritten = getattr(self, rule)(*args, **hints)
    333 
    334                 if rewritten is not None:

/home/sean/sympy/sympy/physics/quantum/spin.py in _eval_rewrite_as_Jz(self, *args, **options)
    903         if isinstance(self, Bra):
    904             return self._rewrite_basis(Jz, JzBra, **options)
--> 905         return self._rewrite_basis(Jz, JzKet, **options)
    906 
    907     def _rewrite_basis(self, basis, evect, **options):

/home/sean/sympy/sympy/physics/quantum/spin.py in _rewrite_basis(self, basis, evect, **options)
    943             else:
    944                 state = evect(j, mi, *args)
--> 945                 lt = Rotation.D(j, mi, self.m, *angles)
    946                 return Sum(lt * state, (mi,-j,j))
    947 

TypeError: D() takes exactly 7 arguments (4 given)
```

This really comes down to addressing the 'TODO: better way to get angles of rotation' in _rewrite_basis and 'TODO: Use KroneckerDelta if all Euler angles == 0' in _represent_base.

A good test for this is:

```
qapply(Jz * JzKet(1, m)) == -hbar * KroneckerDelta(-1, m) * JzKet(1, -1) + hbar * KroneckerDelta(1, m) * JzKet(1, 1)
```

while not causing any of the rewrite/represent tests to fail.

### PR DESCRIPTION
**Pull request by VBhat97**
#### References to other Issues or PRs
Fixes #6521 

#### Brief description of what is fixed or changed
Added conditional statement in _eval_wignerd of spin.py file to return KroneckerDelta when alpha,beta and gamme are 0.
Added new function named test_wignerD to test these cases in test_spin.py file.

#### Release Notes

* physics.quantum
  * WignerD now evaluates to KroneckerDelta in some cases.

### CODE REVIEW

File: sympy/physics/quantum/spin.py [hunk 1]
```diff
@@ -856,39 +856,42 @@ def _eval_wignerd(self):
...
-            # alpha=gamma=0.
-            for k in range(2*j + 1):
-                if k > j + mp or k > j - m or k < mp - m:
-                    continue
-                r += (S.NegativeOne)**k*binomial(j + mp, k)*binomial(j - mp, k + m - mp)
-            r *= (S.NegativeOne)**(m - mp) / 2**j*sqrt(factorial(j + m) *
-                    factorial(j - m) / (factorial(j + mp)*factorial(j - mp)))
+        if alpha==0 and beta==0 and gamma==0:
+            return KroneckerDelta(m,mp)
         else:
```
Comment (oscarbenjamin) [inline sympy/physics/quantum/spin.py:861]: It isn't necessary to indent everything below the `if` here (the else clause isn't needed).

Comment (oscarbenjamin) [reply sympy/physics/quantum/spin.py:861]: Also use space after comma and either side of `==`

Comment (VBhat97) [reply sympy/physics/quantum/spin.py]: Hey @oscarbenjamin !  Worked on the comments and updated the PR!
Thanks for the comments!

File: sympy/physics/quantum/tests/test_spin.py [hunk 2]
```diff
@@ -3675,6 +3675,14 @@ def test_wignerd():
         j, m, mp, alpha, beta, gamma) == WignerD(j, m, mp, alpha, beta, gamma)
     assert Rotation.d(j, m, mp, beta) == WignerD(j, m, mp, 0, beta, 0)
 
+def test_wignerD():
+    i,j=symbols('i j')
+    assert Rotation.D(1,1,1,0,0,0) == WignerD(1,1,1,0,0,0)
+    assert Rotation.D(1,1,2,0,0,0) == WignerD(1,1,2,0,0,0)
+    assert Rotation.D(1,i**2 - j**2, i**2 - j**2,0,0,0) == WignerD(1,i**2 - j**2, i**2 - j**2,0,0,0)
```
Comment (oscarbenjamin) [inline sympy/physics/quantum/tests/test_spin.py:3682]: Use a space after a comma

**Comment by oscarbenjamin** [comment]
I only have a minor formatting comment. I'm focussing on code quality but I'm less knowledgable about the domain here (it's a while since I had anything to do with QM).

@m93a @flacjacket can either of you review this? (It's a fairly small change)

File: sympy/physics/quantum/tests/test_spin.py (hunk 2 above)
Comment (VBhat97) [reply sympy/physics/quantum/tests/test_spin.py]: Ohh yep! Worked on these comments and updated! 

Thanks!

**Review by flacjacket — APPROVED** [review]
This looks good to me, thanks for helping to clear up my 9 year old TODO!

**Comment by oscarbenjamin** [comment]
Thanks both!

**Comment by VBhat97** [comment]
Thank you for your help!

**Pull request merged**